tests/test_archive.py

```diff
@@ -1163,3 +1163,12 @@ def test_compress_win32_absolute_arcname(tmp_path):
     #
     p7zip_test(tmp_path / "target.7z")
     libarchive_extract(tmp_path / "target.7z", tmp_path.joinpath("tgt2"))
+
+
+@pytest.mark.files
+def test_empty_stream(tmp_path):
+    try:
+        with py7zr.SevenZipFile(file="test.7z", mode="w") as archive:
+            archive.writestr(data="", arcname="empty.txt")
+    except AttributeError:
+        pytest.fail("Exception raised")
```

Comment:
Please add several assertions to check resulted archives.
There is not necessary to use try except block.

please learn from other tests. Please do not use `with` syntax but you can make test like

```
archive = py7zr.SevenZipFile(tmp_path / "test.7z", "w")
archive.writestr(data="", arcname="empty.txt")
# assertions here, following check archive has a ONE file
assert len(archive.files.files_list) == 1
# When call close(), SevenZipFile write data into disk.
archive.close()
# you can extract file with py7zr  here and check whether created file size is zero.
#
# check a created archive. try extract by p7zip command
p7zip_test(tmp_path / "test.7z")
```